Xamarin.Forms.Core/VisualElement_StyleSheet.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.IO;
+using System.Reflection;
+
+using Xamarin.Forms.Internals;
+using Xamarin.Forms.StyleSheets;
+using Xamarin.Forms.Xaml;
+
+namespace Xamarin.Forms
+{
+	public partial class VisualElement : IStyleSelectable, IStylable
+	{
+		IList<string> IStyleSelectable.Classes
+			=> StyleClass;
+
+		public static readonly BindableProperty StyleSheetProperty =
+			BindableProperty.Create("StyleSheet", typeof(StyleSheet), typeof(VisualElement), default(StyleSheet),
+				propertyChanged: (bp, o, n) => ((VisualElement)bp).OnStyleSheetChanged((StyleSheet)o, (StyleSheet)n));
+
+		[TypeConverter(typeof(StyleSheetConverter))]
+		public StyleSheet StyleSheet {
```

Comment:
there's no actual public way to create a StyleSheet, so this expects a string uri in Xaml and the StyleSheet will be generated